core/pom.xml

```diff
@@ -614,6 +614,12 @@ THE SOFTWARE.
       <artifactId>jzlib</artifactId>
       <version>1.1.3-kohsuke-1</version>
     </dependency>
+    <!-- Should be removed after splitting maven-builder-plugin see JENKINS-26462 -->
+    <dependency>
```

Comment:
I suppose `ProjectWithMaven` needs to be deprecated and converted into an extension point that allows any code to declare what `MavenInstallation` is associated with an item. Then [this](https://github.com/jenkinsci/jenkins/blob/9fce1ee933eb5276baff977d562fc8e183f1c8d6/core/src/main/java/hudson/model/Project.java#L66), [this](https://github.com/jenkinsci/maven-plugin/blob/619fb44aa2c29edce9a6c3c74ec055dedd162630/src/main/java/hudson/maven/AbstractMavenProject.java#L51), and [this](https://github.com/jenkinsci/maven-plugin/blob/619fb44aa2c29edce9a6c3c74ec055dedd162630/src/main/java/hudson/maven/MavenUtil.java#L76) need to be fixed to call the new API.